src/blog/jwt-identification-and-hot-streams/index.md

```diff
@@ -0,0 +1,172 @@
+
+To sum up, all we need from the `#connect` method is to accept-or-reject connection and populate identifiers. And we can encapsulate this logic within a single [JWT][] token!
+
+This is how it could look in pure Ruby. First, let's create a helper to build a connection url with a token:
+
+```ruby
+class ApplicationHelper
+  def action_cable_with_jwt_meta_tag(**identifiers)
+    # Token TTL (5 minutes)
+    exp = Time.now.to_i + 60*5
+    # JWT payload.
+    # The #serialize_identifiers is responsible for serializing objects into strings
+    # (AnyCable uses GlobalID for that)
+    payload = {ext: serialize_identifiers(identifiers), exp: exp}
+
+    token = JWT.encode payload, JWT_ENCRYPTION_KEY, "HS256"
+
+    base_url = ActionCable.server.config.url || ActionCable.server.config.mount_path
+    tag "meta", name: "action-cable-url", content: "#{base_url}?token=#{token}"
+  end
+end
+```
+
+Now we can add it to our layout:
+
+```erb
+<%= action_cable_with_jwt_meta_tag(current_user: current_user) %> would render:
+# => <meta name="action-cable-url" content="ws://demo.anycable.io/cable?token=eyJhbGciOiJIUzI1NiJ9....EWCEzziOx3sKyMoNzBt20a3QvhEdxJXCXaZsA-f-UzU" />
+```
+
+Finally, we need to verify the token and extract the identifiers in the `#connect` method:
+
+```ruby
+module ApplicationCable
+  class Connection < ActionCable::Connection::Base
+    identified_by :current_user
+
+    def connect
+      token = request.params[:token]
+
+      decoded_token = JWT.decode token, JWT_ENCRYPTION_KEY, true, { algorithm: "HS256" }
+      # The #deserialize_identifiers method is a counterpart to #serialize_identifiers
+      # and is responsible for converting strings back to objects
+      identifiers = deserialize_identifiers(JSON.parse(decoded_token["ext"]))
+
+      # Populate identifiers
+      identifiers.each do |k, v|
+        self.public_send("#{k}=", v)
+      end
+    rescue JWT::DecodeError
+      reject_unauthorized_connection
+    end
+  end
+end
+```
+
+Doesn't look like a simple solution 😕 If you'll continue staring at this piece of code for a while, you may notice that the implementation has nothing specific to a particular application: we retrieve a token, verify it and extract identifiers. If so, why can't we re-implement this in Go? We can!
+
+This is how AnyCable Go PRO JWT identification works:
+
+- A client passes a token along the connection request (via a query string or a header).
+- Go server retrieves a header, verifies its signature and TTL, and rejects the connection if token is invalid.
+- If token is valid, we extract identifiers, store them in the connection metadata and finalize the handshake (by sending a `{"type":"welcome"}` message).
```

Comment:
If the token is valid, we extract the identifiers, store them in the connection metadata and finalize the handshake (by sending a `{"type":"welcome"}` message).